pkg/cephfs/nodeserver.go

```diff
@@ -308,6 +309,89 @@ func (ns *NodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstag
 	return &csi.NodeUnstageVolumeResponse{}, nil
 }
 
+func (ns *NodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
+
+	var err error
+	targetPath := req.GetVolumePath()
+	if targetPath == "" {
+		err = fmt.Errorf("targetpath %v is empty", targetPath)
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	volID := req.GetVolumeId()
+	/* TODO: Map the volumeID to the targetpath.
+
+	we need secret to connect to the ceph cluster to get the volumeID from volume
+	Name, however `secret` field/option is not available  in NodeGetVolumeStats spec,
+	Below issue covers this request and once its available, we can do the validation
+	as per the spec.
+	https://github.com/container-storage-interface/spec/issues/371
+
+	*/
+
+	isMnt, err := isMountPoint(targetPath)
+
+	if err != nil {
```

Comment:
suggestion
```
if err != nil{
     if os.IsNotExist(err){
         return nil, status.Error(codes.InvalidArgument, err.Error())
      }
     else{
           return nil, status.Error(codes.Internal, err.Error())
       }
}
		klog.Errorf("stat failed on path: %v, error: %v", targetPath, err)
```

Reply:
isMountPoint is always returning `status.Error(codes.Internal, err.Error())` on errors, but the current error message does need some massaging around "stat failing etc."

Reply:
humble can you just return an error here?